### PR TITLE
subsys : shell : Added API to change shell prompt

### PR DESCRIPTION
**Pull request by jakub-uC**
1. Added API to change shell prompt in runtime.
2. Added prompt buffer length configuration in Kconfig.

Signed-off-by: Jakub Rzeszutko <jakub.rzeszutko@nordicsemi.no>

### CODE REVIEW

File: subsys/shell/shell.c [hunk 1]
```diff
@@ -1145,7 +1145,7 @@ static int shell_instance_init(const struct shell *shell, const void *p_config,
 			       bool use_colors)
 {
 	assert(shell);
-	assert(shell->ctx && shell->iface && shell->name);
+	assert(shell->ctx && shell->iface && shell->prompt);
```
Comment (carlescufi) [inline subsys/shell/shell.c]: what `assert` is this? I thought we used `_ASSERT()`

Comment (carlescufi) [reply subsys/shell/shell.c]: I also see `_Static_assert` but we already have: `BUILD_ASSERT`

Comment (carlescufi) [reply subsys/shell/shell.c]: If those are defined by the shell subsystem, could you switch to the standard Zephyr ones?

File: subsys/shell/shell.c [hunk 2]
```diff
@@ -1663,6 +1663,21 @@ void shell_help_print(const struct shell *shell,
 	help_subcmd_print(shell);
 }
 
+int shell_prompt_change(const struct shell *shell, char *prompt)
+{
+
+	size_t len = shell_strlen(prompt);
```
Comment (nordic-krch) [inline subsys/shell/shell.c]: isn't strlen returning length without \0. If so then if check should be
`len < CONFIG...` and memcpy `len+1`

File: subsys/shell/shell.c (hunk 1 above)
Comment (jarz-nordic) [reply subsys/shell/shell.c]: I am using assert from here:
zephyr/lib/libc/minimal/include/assert.h
Is there any other recommended?

Comment (jarz-nordic) [reply subsys/shell/shell.c]: I see that `assert` is the same as `__ASSERT_NO_MSG`

File: subsys/shell/shell.c (hunk 2 above)
Comment (jarz-nordic) [reply subsys/shell/shell.c]: It is fine, prompt table is defined as: `[CONFIG_SHELL_PROMPT_LENGTH + 1]`

Comment (nordic-krch) [reply subsys/shell/shell.c]: ok, but then you are missing setting termination ('\0'). You should have `len+1` in memcpy

Comment (jarz-nordic) [reply subsys/shell/shell.c]: @nordic-krch : done

**Review by nordic-krch — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**